Polyfills/Canvas/Source/Canvas.cpp

```diff
@@ -41,11 +42,15 @@ namespace Babylon::Polyfills::Internal
 
     void NativeCanvas::LoadTTF(const Napi::CallbackInfo& info)
     {
-        std::string fontName = info[0].As<Napi::String>().Utf8Value();
+        auto& graphicsImpl{Babylon::GraphicsImpl::GetFromJavaScript(info.Env())};
+        const auto cancellationSource = std::shared_ptr<arcana::cancellation_source>();
         const auto buffer = info[1].As<Napi::ArrayBuffer>();
-
-        fontsInfos[fontName] = std::vector<uint8_t>(buffer.ByteLength());
-        memcpy(fontsInfos[fontName].data(), (uint8_t*)buffer.Data(), buffer.ByteLength());
+        std::vector<uint8_t> fontBuffer{};
+        fontBuffer.resize(buffer.ByteLength());
+        memcpy(fontBuffer.data(), (uint8_t*)buffer.Data(), buffer.ByteLength());
+        arcana::make_task(graphicsImpl.BeforeRenderScheduler(), arcana::cancellation::none(), [fontName{ info[0].As<Napi::String>().Utf8Value() }, fontData{ std::move(fontBuffer) }]() {
```

Comment:
If this is going to be async, then this function should return a promise right?

Reply:
if i understand correctly this function doesn't need to resolve before declaring text. Given returning a promise would require updating babylon.js, i'm inclined to say that returning a promise here is beyond the scope of this review and should be addressed when we finalize the public api contract for loading fonts (fontface, etc.)

Reply:
actually loadTTF isn't defined anywhere public. let me return a promise here.

Reply:
this should now be fixed